url90b6689baa_adsonrocha_PD321_tgz-pJ8-MyFuzzyLiteClassJ8/src/com/fuzzylite/imex/FclExporter.java

```diff
@@ -0,0 +1,296 @@
+        }
+        result.append("END_DEFUZZIFY\n");
+        return result.toString();
+    }
+
+    public String toString(RuleBlock ruleBlock) {
+        StringBuilder result = new StringBuilder();
+        result.append(String.format("RULEBLOCK %s\n", ruleBlock.getName()));
+        if (!ruleBlock.isEnabled()) {
+            result.append(String.format(indent + "ENABLED : %s;\n",
+                    String.valueOf(ruleBlock.isEnabled()).toUpperCase()));
+        }
+        if (ruleBlock.getConjunction() != null) {
+            result.append(String.format(indent + "AND : %s;\n", toString(ruleBlock.getConjunction())));
+        }
+        if (ruleBlock.getDisjunction() != null) {
+            result.append(String.format(indent + "OR : %s;\n", toString(ruleBlock.getDisjunction())));
+        }
+        if (ruleBlock.getActivation() != null) {
+            result.append(String.format(indent + "ACT : %s;\n", toString(ruleBlock.getActivation())));
+        }
+
+        int index = 1;
+        for (Rule rule : ruleBlock.getRules()) {
+            result.append(String.format(indent + "RULE %d : %s\n", index++, rule.getText()));
+        }
+        result.append("END_RULEBLOCK\n");
+        return result.toString();
+    }
+
+    public String toString(Term term) {
+        if (term == null) {
+            return "";
+        }
+
+        if (term instanceof Discrete) {
+            StringBuilder result = new StringBuilder();
+            Discrete discrete = (Discrete) term;
+            for (int i = 0; i < discrete.x.size(); ++i) {
+                result.append(String.format("(%s, %s)",
+                        Op.str(discrete.x.get(i)), Op.str(discrete.y.get(i))));
+                if (i + 1 < discrete.x.size()) {
+                    result.append(" ");
+                }
+            }
+            return result.toString();
+        }
+
+        if (term instanceof Constant) {
+            Constant constant = (Constant) term;
+            return Op.str(constant.getValue());
+        }
+
+        return term.getClass().getSimpleName() + " " + term.parameters();
+    }
+
+    public String toString(Defuzzifier defuzzifier) {
+        if (defuzzifier == null) {
+            return "";
+        }
+        if (defuzzifier instanceof Centroid) {
+            return "COG";
+        }
+        if (defuzzifier instanceof Bisector) {
+            return "COA";
+        }
+        if (defuzzifier instanceof SmallestOfMaximum) {
+            return "LM";
+        }
+        if (defuzzifier instanceof LargestOfMaximum) {
+            return "RM";
+        }
+        if (defuzzifier instanceof MeanOfMaximum) {
+            return "MM";
+        }
+        if (defuzzifier instanceof WeightedAverage) {
+            return "COGS";
+        }
+        if (defuzzifier instanceof WeightedSum) {
+            return "COGSS";
+        }
+        return defuzzifier.getClass().getSimpleName();
+    }
+
+    public String toString(Norm norm) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.